lib/artoo/robot.rb

```diff
@@ -186,23 +186,29 @@ def respond_to_missing?(method_name, include_private = false)
 
     def initialize_connections(params={})
       @connections = {}
-      connection_types.each {|ct|
-        Logger.info "Initializing connection #{ct[:name].to_s}..."
-        cp = params[ct[:name]] || {}
-        c = Connection.new(ct.merge(cp).merge(:parent => current_instance))
-        @connections[ct[:name]] = c
-      }
+      connection_types.each {|type| initialize_connection(params, type, @connections)}
     end
 
     def initialize_devices(params={})
       @devices = {}
-      device_types.each {|d|
-        Logger.info "Initializing device #{d[:name].to_s}..."
-        dp = params[d[:name]] || {}
-        d = Device.new(d.merge(dp).merge(:parent => current_instance))
-        instance_eval("def #{d.name}; return devices[:#{d.name}]; end")
-        @devices[d.name.intern] = d
-      }
+      device_types.each {|type| initialize_device(params, type, @devices)}
+    end
+
+    def initialize_connection(params, type, connections)
+      connection = initialize_object(Connection, params, type)
+      @connections[type[:name]] = connection
+    end
+
+    def initialize_device(params, type, devices)
+      type = initialize_object(Device, params, type)
+      instance_eval("def #{type.name}; return devices[:#{type.name}]; end")
+      @devices[type.name.intern] = type
+    end
+
+    def initialize_object(klass, params, type)
+      Logger.info "Initializing device #{type[:name].to_s}..."
```

Comment:
Now that this is being used to initialize both connections and devices, it should probably be something like `Logger.info "Initializing #{klass.to_s} #{type[:name].to_s}..."`